statement.c

```diff
@@ -2319,6 +2319,10 @@ MYLOG(DETAIL_LOG_LEVEL, "!!SC_fetch return =%d\n", ret);
 					}
 
 					STR_TO_NAME(self->cursor_name, QR_get_value_backend_text(rhold.first, 0, i));
+					/* Skip NULL refcursors to support a variable number of results */
+					if (!SC_cursor_is_valid(self))
+						continue;
```

Comment:
This prevents the driver attempting to fetch a NULL refcursor, which results in an error. This change allows procedures to return optional result sets, e.g.

```plpgsql
create or replace procedure refproc
(in num_cursor integer, inout ref1 refcursor default null, inout ref2 refcursor default null) as
$procedure$
DECLARE
BEGIN
    OPEN ref1 FOR SELECT id, t FROM testtab1 ORDER BY id ASC;
    IF num_cursor = 2 THEN
        OPEN ref2 FOR SELECT t, id FROM testtab1 ORDER BY id DESC;
    END IF;
END;
$procedure$
LANGUAGE plpgsql
```

Reply:
I thought about making this `continue` a `break` instead, because procedures would usually only return `NULL` refcursors for the right-most parameters to avoid the caller getting confused about which result set is which. But I did not want to be too opinionated and force this convention, as there may be other cases to consider.

For example, the Amazon Schema Conversion Tool is known to produce procedures which logically should only have one refcursor parameter, but instead has multiple where only one is ever used at a time (depending on which branch of code is executed).